tests/Kros.KORM.UnitTests/Query/DbSetShould.cs

```diff
@@ -237,6 +237,33 @@ public void CommitPendingChanges()
             dbSet.DeletedItems.Should().BeEmpty();
         }
 
+        [Fact]
+        public void ThrowExceptionWhenProviderDoesNotSupportIdnetity()
+        {
+            var tableInfo = new TableInfo(new List<ColumnInfo>()
+            {
+                new ColumnInfo(){
+                    Name = "Id",
+                    IsPrimaryKey = true,
+                    AutoIncrementMethodType = AutoIncrementMethodType.Indetity
+                }
+            }, new List<PropertyInfo>(), null);
+
+            IDbSet<Person> dbSet = new DbSet<Person>(Substitute.For<ICommandGenerator<Person>>(),
+                                                     new FakeProvider(),
+                                                     Substitute.For<IQuery<Person>>(),
+                                                     tableInfo);
+
+            dbSet.Add(new Person() { Name = "A" });
+
+            var action = new Action(() => dbSet.CommitChanges());
+
+            action.Should()
+                .Throw<InvalidOperationException>()
+                .WithMessage("The provider 'FakeProvider' does not support inserting values " +
```

Comment:
I wouldn't check for whole message. FluentAssertions supports _pattern matching_ so we can just check specific words: `"*FakeProvider*Person*"`